### ISSUE TITLE
Filesystem Manager

### ISSUE THREAD

**Issue by the-maldridge** [the-maldridge/vInstaller#3]
The installer needs a filesystem configuration manager.  This needs to know a set of recipes for filesystems that can be "filled in" by the installer frontends, but should also accept a configuration being pushed back to it.

**Comment by pltrz** [comment]
How should the i/o and structure for this program look?

Would it be something along the lines of the following?

0.  User should have already partitioned the drive(s) in a previous step.

1.  Show user list of current partitions.

2.  User picks a partition like `/dev/sdaX`

    2.1 Should the partition be mounted to a certain mountpoint? 
    - perhaps `/` first, since it's the only *mandatory* one and optionally others after, like `/boot`, `/home` etc.

    2.2 Should the partition be formatted?
    - list filesystems such as `ext4`, `vfat` etc.

3.  List current filesystems and mountpoints for the new system, as it looks so far.
    Give the user an option to go back and do numbers `1` and `2` again and add another partition to the scheme.

5.  If user decides everything looks OK in number `3`, save this scheme to a `struct` or something and, ready to be executed after?

Let me know if this is more or less what it should be and any other details.

**Comment by the-maldridge** [comment]
Right, so this is more or less accurate.  What I envision is that (for now) the installer bails out to cfdisk as expected, or offers you the choice to use an entire disk.  Either way it then offers you menus of disk layouts similar to debian's traditional installer.  If you picked a whole disk to use, it then uses the default layout scheme for that layout, maybe allowing you to adjust sizes of individual partitions.  If the user pre-partitions the disk, then they should be able to map partitions from the disk to partitions in a menu selection.

The hard one is the fully custom layout, which involves being able to say what partitions go with what mountpoints without knowing in advance the number of partitions.

Architecturally, I see this working like the frontends with a single package that controls the loading of them and different implementations.  The difference here being that each implementation is a different  disk layout.
